packages/service-library/src/servicelib/archiving_utils/_interface_7zip.py

```diff
@@ -0,0 +1,330 @@
+            self.finished = True
+
+    async def parse_chunk(self, chunk: str) -> None:
+        self._prase_progress(chunk)
+
+        if self.total_bytes is not None and self.percent is not None:
+            # total bytes decompressed
+            current_bytes_progress = int(self.percent * self.total_bytes / 100)
+
+            # only emit an update if something changed since before
+            bytes_diff = current_bytes_progress - self.emitted_total
+            if self.emitted_total == 0 or bytes_diff > 0:
+                await self.decompressed_bytes(bytes_diff)
+
+            self.emitted_total = current_bytes_progress
+
+        # if finished emit the remaining diff
+        if self.total_bytes and self.finished and not self.finished_emitted:
+
+            await self.decompressed_bytes(self.total_bytes - self.emitted_total)
+            self.finished_emitted = True
+
+
+async def _stream_output_reader(
+    stream: asyncio.StreamReader,
+    *,
+    output_handlers: list[Callable[[str], Awaitable[None]]] | None,
+    chunk_size: NonNegativeInt = 16,
+    lookbehind_buffer_size: NonNegativeInt = 40,
+) -> str:
+    # NOTE: content is not read line by line but chunk by chunk to avoid missing progress updates
+    # small chunks are read and of size `chunk_size` and a bigger chunk of
+    # size `lookbehind_buffer_size` + `chunk_size` is emitted
+    # The goal is to not split any important search in half, thus giving a change to the
+    #  `output_handlers` to properly handle it
+
+    # NOTE: at the time of writing this, the biggest possible thing to capture search would be:
+    # ~`9.1TiB` -> literally ` 9999999999999 bytes ` equal to 21 characters to capture,
+    # with the above defaults we the "emitted chunk" is more than double in size
+    # There are no foreseeable issued due to the size of inputs to be captured.
+
+    if output_handlers is None:
+        output_handlers = []
+
+    command_output = ""
+    lookbehind_buffer = ""
+
+    while True:
+        read_chunk = await stream.read(chunk_size)
+
+        if not read_chunk:
+            # process remaining buffer if any
+            if lookbehind_buffer:
+                await asyncio.gather(
+                    *[handler(lookbehind_buffer) for handler in output_handlers]
+                )
+            break
+
+        # `errors=replace`: avoids getting stuck when can't parse utf-8
+        chunk = read_chunk.decode("utf-8", errors="replace")
+
+        command_output += chunk
+        chunk_to_emit = lookbehind_buffer + chunk
+        lookbehind_buffer = chunk_to_emit[-lookbehind_buffer_size:]
+
+        await asyncio.gather(*[handler(chunk_to_emit) for handler in output_handlers])
+
+    return command_output
+
+
+async def _run_cli_command(
+    command: str,
+    *,
+    output_handlers: list[Callable[[str], Awaitable[None]]] | None = None,
+) -> str:
+    """
+    Raises:
+        ArchiveError: when it fails to execute the command
```

Comment:
don't you already have something for running async_command in the sidecar  no? would it not make sense to re-use this?